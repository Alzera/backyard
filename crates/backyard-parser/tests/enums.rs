use backyard_parser::parse;

#[test]
fn basic() {
  let asts = parse(true, "enum Suit {
  case Hearts;
  case Spades;
}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn typed() {
  let asts = parse(true, "enum Suit: int {
  case Hearts = 5;
  case Spades = 6;
}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn content() {
  let asts = parse(
    true,
    "enum Suit implements SuitInterface {
  case Hearts;
  case Spades;

  public const MY_CONST = \"constant\";

  public function color(): string {
    return match($this) {
      Suit::Hearts, Suit::Diamonds => 'Red',
      Suit::Clubs, Suit::Spades => 'Black'
    };
  }
}"
  ).unwrap();
  insta::assert_yaml_snapshot!(asts);
}
