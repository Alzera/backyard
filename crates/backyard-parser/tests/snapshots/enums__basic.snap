---
source: crates/backyard-parser/tests/enums.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: enum
    name:
      node_type: identifier
      name: Suit
      loc:
        start:
          line: 1
          column: 5
          offset: 5
        end:
          line: 1
          column: 9
          offset: 9
      leadings: ~
      trailings: ~
    enum_type: ~
    implements: ~
    body:
      - node_type: enum_item
        value:
          node_type: identifier
          name: Hearts
          loc:
            start:
              line: 2
              column: 7
              offset: 19
            end:
              line: 2
              column: 13
              offset: 25
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 2
            column: 2
            offset: 14
          end:
            line: 2
            column: 7
            offset: 19
        leadings: ~
        trailings: ~
      - node_type: enum_item
        value:
          node_type: identifier
          name: Spades
          loc:
            start:
              line: 3
              column: 7
              offset: 34
            end:
              line: 3
              column: 13
              offset: 40
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 3
            column: 2
            offset: 29
          end:
            line: 3
            column: 7
            offset: 34
        leadings: ~
        trailings: ~
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 4
        column: 0
        offset: 42
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 4
    column: 0
    offset: 42
leadings: ~
trailings: ~
